server/athenian/api/controllers/goal_controller.py

```diff
@@ -168,4 +182,60 @@ async def parse_request_repositories(
 
 async def measure_goals(request: AthenianWebRequest, body: dict) -> web.Response:
     """Calculate the metrics for the goal tree."""
-    raise NotImplementedError()
+    goals_request = AlignGoalsRequest.from_dict(body)
+    team = goals_request.team
+    team_rows, meta_ids, jira_config = await gather(
+        fetch_teams_recursively(
+            goals_request.account,
+            request.sdb,
+            select_entities=(Team.id, Team.name, Team.members, Team.parent_id),
+            # teamId 0 means to implicitly use the single root team
+            root_team_ids=None if team == 0 else [team],
+        ),
+        get_metadata_account_ids(goals_request.account, request.sdb, request.cache),
+        get_jira_installation_or_none(
+            goals_request.account, request.sdb, request.mdb, request.cache,
+        ),
+    )
+    team_tree = build_team_tree_from_rows(team_rows, None if team == 0 else team)
+    team_member_map = flatten_teams(team_rows)
+
+    team_ids = [row[Team.id.name] for row in team_rows]
+    team_goal_rows, prefixer = await gather(
+        fetch_team_goals(goals_request.account, team_ids, request.sdb),
```

Comment:
Please add a comment in `fetch_team_goals` that the order is mandatory. Otherwise, `groupby` will silently betray us.